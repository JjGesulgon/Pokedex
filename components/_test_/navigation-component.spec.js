import { shallowMount } from "@vue/test-utils";
import NavigationComponent from "../navigation-component";


const factory = () => {
  return shallowMount(NavigationComponent, {
  });
};

describe("NavigationComponent", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});