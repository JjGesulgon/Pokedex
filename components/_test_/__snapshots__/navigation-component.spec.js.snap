// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationComponent renders properly 1`] = `
<nav class="nav-container absolute z-50 sticky top-0 flex items-center justify-between flex-wrap bg-red-700 p-6 text-white">
  <div class="flex items-center flex-shrink-0 mr-6"><img src="/pokeball.svg" alt="book lover" class="w-10">
    &nbsp;
    <span class="font-semibold text-lg tracking-tight">Pokedex</span>
  </div>
  <div class="block lg:hidden"><button class="flex px-3 py-2 border rounded border-black focus:outline-none"><svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg" class="fill-current h-3 w-3">
        <title>Menu</title>
        <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"></path>
      </svg></button></div>
  <div class="w-full block flex-grow lg:flex lg:items-center lg:w-auto hidden">
    <div class="text-sm lg:flex-grow"></div>
    <div class="tracking-tight">
      <form>
        <n-link to="/" class="block mt-4 lg:inline-block lg:mt-0 hover:underline mr-4 focus:outline-none">
          Pokemon List
        </n-link> <input type="search" name="search" placeholder="Search" class="border-2 border-gray-300 bg-white h-10 pl-2 pr-8 rounded-lg text-black text-sm focus:outline-none">
      </form>
    </div>
  </div>
</nav>
`;
